File: examples/python-client/log.py
import requests

class LoggingClient:
    BASE_URL = "http://localhost:3005/post/log"

    def log(self, service_name, log_level, message, user_id=None):
        """
        Send a log message to the logging service.
        :param service_name: Name of the service generating the log.
        :param log_level: Log level (e.g., INFO, ERROR).
        :param message: Log message content.
        :param user_id: User ID for user-specific logs. None for user-independent logs.
        """
        payload = {
            "service_name": service_name,
            "log_level": log_level,
            "message": message
        }
        if user_id:
            payload["user_id"] = user_id

        try:
            response = requests.post(self.BASE_URL, json=payload)
            response.raise_for_status()
            print(f"Log successfully sent: {response.json()}")
        except requests.exceptions.RequestException as e:
            print(f"Failed to send log: {e}")

deneme = LoggingClient()
deneme.log(log_level="INFO",message="Message Log",service_name="Example_Service",user_id=3)

File: examples/js-client/log.js
class LoggingClient {
  constructor() {
    this.BASE_URL = "http://localhost:3005/post/log";
  }

  async log({ service_name, log_level, message, user_id = null }) {
    const payload = {
      service_name,
      log_level,
      message,
    };

    if (user_id !== null) {
      payload.user_id = user_id;
    }

    try {
      const response = await fetch(this.BASE_URL, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(payload),
      });

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
      }

      const data = await response.json();
      console.log("Log successfully sent:", data);
    } catch (error) {
      console.error("Failed to send log:", error);
    }
  }
}

// Tested and worked. Open JS CLIENT comment line in the index.html to try


// Example
const logger = new LoggingClient();
logger.log({
  service_name: "Desktop-App",
  log_level: "INFO",
  message: "DENEME LOG MESAJ",
  user_id: 3,
});
